Depth and ms8537

FlorianPix opened the pull request:
## Describe problem solved by this pull request
fixes #20049
fixes #20051
fixes #20073 

## Test 
Tested on a pixhawk4 with a MS5837-30BA in a 1.5m deep fresh water pool.
Accuracy ±1cm

FlorianPix committed:
fixed undeclared variable in MS5837 driver: closes #20049;
fixed wrong unit in MS5837.cpp: closes #20051;

FlorianPix committed:
added altitude/depth calculation for water; added parameter BARO_MEDIUM to differentiate between air, fresh water and salt water in depth calculation; closes #20073

bresch commented on src/modules/sensors/vehicle_air_data/params.c:
```diff
@@ -53,3 +53,12 @@ PARAM_DEFINE_FLOAT(SENS_BARO_QNH, 1013.25f);
+ * @max 2
+ * @group Sensors
+ */
+PARAM_DEFINE_INT32(BARO_MEDIUM, 0);
```
```suggestion
PARAM_DEFINE_INT32(SENS_BARO_MEDIUM, 0);
```

FlorianPix replied:
3670f4722d16f331a9b23a5c6910b4a5c062fdea

bresch commented on src/modules/sensors/vehicle_air_data/params.c:
```diff
@@ -53,3 +53,12 @@ PARAM_DEFINE_FLOAT(SENS_BARO_QNH, 1013.25f);
+ * air, fresh or salt water
+ *
+ * @min 0
+ * @max 2
```
You can add the meaning of 0,1,2 in an enum like this: https://github.com/PX4/PX4-Autopilot/blob/cfc579542e1bad253c26fdc90fb432baf5ffcd07/src/drivers/gps/params.c#L43-L47

FlorianPix replied:
3670f4722d16f331a9b23a5c6910b4a5c062fdea

bresch commented on src/modules/sensors/vehicle_air_data/VehicleAirData.cpp:
```diff
@@ -297,27 +297,66 @@ void VehicleAirData::Run()
-
+    float altitude = 0.0f;
+
+    if (_param_baro_medium.get() == 0)
```
you could add an enum in the header file like that to make it more readable
```
enum class Medium {
   Air,
   FreshWater,
   SaltWater
}
```
```suggestion
    if (_param_baro_medium.get() == Medium::Air)
```

FlorianPix replied:
3670f4722d16f331a9b23a5c6910b4a5c062fdea

Jaeyoung-Lim reviewed (commented):
Great!

Would be cool if we can propagate this to the simulated ROV setups: 
- SDF file: https://github.com/PX4/PX4-SITL_gazebo/blob/bd941689b6e9d3e628a1e786d1d042d48005dc7b/models/uuv_bluerov2_heavy/uuv_bluerov2_heavy.sdf#L539-L543
- pressure calculations: https://github.com/PX4/PX4-SITL_gazebo/blob/bd941689b6e9d3e628a1e786d1d042d48005dc7b/src/gazebo_barometer_plugin.cpp#L148-L157

FlorianPix committed:
added Enum for medium (air, fresh or salt water) to VehicleAirData;
changed param name BARO_MEDIUM to SENS_BARO_MEDIUM;
added value comments to param SENS_BARO_MEDIUM;

dagar commented:
I don't particularly like that we're doing this within "air data", but it seems relative inoffensive so far.

FlorianPix commented:
@spiderkeys had some interesting thoughts on #20073 so you might not want to merge this pull request entirely, but just the sensor bug fixes. I don't work at the same company (for UUVs) anymore so I can't spend time on this at the moment.
